nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/hive/streaming/NiFiRecordSerDe.java

```diff
@@ -187,132 +202,127 @@ private Object extractCurrentField(Record record, RecordField field, TypeInfo fi
                 }
                 switch (primitiveCategory) {
                     case BYTE:
-                        Integer bIntValue = record.getAsInt(fieldName);
-                        val = bIntValue == null ? null : bIntValue.byteValue();
+                        Integer bIntValue = DataTypeUtils.toInteger(fieldValue, field.getDataType().getFormat());
+                        val = bIntValue.byteValue();
                         break;
                     case SHORT:
-                        Integer sIntValue = record.getAsInt(fieldName);
-                        val = sIntValue == null ? null : sIntValue.shortValue();
+                        Integer sIntValue = DataTypeUtils.toInteger(fieldValue, field.getDataType().getFormat());
+                        val = sIntValue.shortValue();
                         break;
                     case INT:
-                        val = record.getAsInt(fieldName);
+                        val = DataTypeUtils.toInteger(fieldValue, field.getDataType().getFormat());
                         break;
                     case LONG:
-                        val = record.getAsLong(fieldName);
+                        val = DataTypeUtils.toLong(fieldValue, field.getDataType().getFormat());
                         break;
                     case BOOLEAN:
-                        val = record.getAsBoolean(fieldName);
+                        val = DataTypeUtils.toBoolean(fieldValue, field.getDataType().getFormat());
                         break;
                     case FLOAT:
-                        val = record.getAsFloat(fieldName);
+                        val = DataTypeUtils.toFloat(fieldValue, field.getDataType().getFormat());
                         break;
                     case DOUBLE:
-                        val = record.getAsDouble(fieldName);
+                        val = DataTypeUtils.toDouble(fieldValue, field.getDataType().getFormat());
                         break;
                     case STRING:
                     case VARCHAR:
                     case CHAR:
-                        val = record.getAsString(fieldName);
+                        val = DataTypeUtils.toString(fieldValue, field.getDataType().getFormat());
                         break;
                     case BINARY:
-                        Object[] array = record.getAsArray(fieldName);
-                        if (array == null) {
-                            return null;
-                        }
+                        Object[] array = DataTypeUtils.toArray(fieldValue, field.getFieldName(), field.getDataType());
```

Comment:
Do we need to get the array element type rather than passing in `field.getDataType()`? The other structured data types (List, e.g.) below do that.

Reply:
Thanks will fix that

Reply:
Actually based on [Hive Binary Design](https://cwiki.apache.org/confluence/display/Hive/Binary+DataType+Proposal) and [Hive Types](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types) does it make sense to only support `ByteArrayRef`, `byte[]` and `String` via getBytes()?